src/status_im2/contexts/chat/messages/pin/events.cljs

```diff
@@ -33,34 +33,46 @@
              (assoc-in [:pagination-info chat-id :all-pin-loaded?]
                        (empty? cursor)))}))
 
-(rf/defn receive-signal
-  [{:keys [db]} pin-messages]
-  (let [{:keys [chat-id]} (first pin-messages)]
-    (when (= chat-id (db :current-chat-id))
-      (let [{:keys [chat-id]}           (first pin-messages)
-            already-loaded-pin-messages (get-in db [:pin-messages chat-id] {})
-            already-loaded-messages     (get-in db [:messages chat-id] {})
-            all-messages                (reduce (fn [acc {:keys [message_id pinned from]}]
-                                                  ;; Add to or remove from pinned message list, and
-                                                  ;; normalizing pinned-by property
-                                                  (let [current-message     (get already-loaded-messages
-                                                                                 message_id)
-                                                        current-message-pin (merge current-message
-                                                                                   {:pinned    pinned
-                                                                                    :pinned-by from})]
-                                                    (cond-> acc
-                                                      (nil? pinned)
-                                                      (dissoc message_id)
+(defn remove-pinned-message
+  [db pinned-message-js]
+  (update-in db
+             [:pin-messages (.-localChatId pinned-message-js)]
+             dissoc
+             (.-message_id pinned-message-js)))
 
-                                                      (and (some? pinned) (some? current-message))
-                                                      (assoc message_id current-message-pin))))
-                                                already-loaded-pin-messages
-                                                pin-messages)]
-        {:db (-> db
-                 (assoc-in [:pin-messages chat-id] all-messages)
-                 (assoc-in [:pin-message-lists chat-id]
-                           (message-list/add-many nil (vals all-messages))))}))))
+(defn add-pinned-message
+  [db pinned-message-js]
+  (let [message-js (.-pinnedMessage pinned-message-js)]
```

Comment:
I believe we prefer to use oops here: https://github.com/status-im/status-mobile/blob/develop/doc/new-guidelines.md#javascript-interop